### PR TITLE
[GLUTEN-8187][VL] Support velox cache metrics

### PR DESCRIPTION
**Pull request by yikf**
## What changes were proposed in this pull request?
Velox backend supports cache functionality controlled by the config spark.gluten.sql.columnar.backend.velox.cacheEnabled, When the cache is enabled, we want to measure its effectiveness through a series of metrics.

This PR intends to add metrics for cache read operations.
## How was this patch tested?
unit tests

### CODE REVIEW

File: backends-velox/src/test/scala/org/apache/gluten/execution/VeloxCacheSuite.scala [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        } else {
+          assert(metrics2("storageReadBytes").value > 0)
+          assert(metrics2("ramReadBytes").value == 0)
+        }
+      } finally {
+        spark.sql(s"DROP TABLE IF EXISTS $table")
+      }
+    }
+
+    Seq("true", "false").foreach(
```
Comment (yikf) [inline backends-velox/src/test/scala/org/apache/gluten/execution/VeloxCacheSuite.scala:90]: this UT will fail due to the current lack of a shutdown, which will be fixed in another PR.

**Comment by yikf** [comment]
@PHILO-HE @zhztheplayer Could you please take a look, thanks!

File: cpp/core/jni/JniWrapper.cc [hunk 2]
```diff
@@ -229,7 +229,10 @@ jint JNI_OnLoad(JavaVM* vm, void* reserved) {
   metricsBuilderClass = createGlobalClassReferenceOrError(env, "Lorg/apache/gluten/metrics/Metrics;");
 
   metricsBuilderConstructor = getMethodIdOrError(
-      env, metricsBuilderClass, "<init>", "([J[J[J[J[J[J[J[J[J[JJ[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J)V");
+      env,
+      metricsBuilderClass,
+      "<init>",
+      "([J[J[J[J[J[J[J[J[J[JJ[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J[J)V");
```
Comment (FelixYBW) [inline cpp/core/jni/JniWrapper.cc:235]: We eventually need a better way to pass metrics.

**Review by FelixYBW — APPROVED** [review]

File: cpp/core/jni/JniWrapper.cc (hunk 2 above)
Comment (zhztheplayer) [reply cpp/core/jni/JniWrapper.cc:235]: Agreed. 

I have similar concern during working on a previous [patch](https://github.com/apache/incubator-gluten/pull/7842#discussion_r1865418672).

The code requires for a reworking.

Comment (yikf) [reply cpp/core/jni/JniWrapper.cc:235]: thanks @FelixYBW , agree with you.

Comment (WangGuangxin) [reply cpp/core/jni/JniWrapper.cc:235]: > We eventually need a better way to pass metrics.

@FelixYBW do you think "Foreign Function and Memory (FFM) API" from JDK 17 is a better way?

Comment (yikf) [reply cpp/core/jni/JniWrapper.cc:235]: @WangGuangxin I investigated the FFM API[1][2] and it seems that there is no way to create and call Java objects from the native side. Can you provide some information on this?

[1] https://docs.oracle.com/en/java/javase/22/core/foreign-function-and-memory-api.html
[2] https://openjdk.org/jeps/454

Comment (yikf) [reply cpp/core/jni/JniWrapper.cc:235]: FFM API/JNI's native side returns two-dimensional arrays, how about parsing mertrics on the java side?

Comment (FelixYBW) [reply cpp/core/jni/JniWrapper.cc:235]: we may also use a protobuf or json string to pass the metrics.
